target/linux/ar71xx/files/arch/mips/ath79/routerboot.c

```diff
@@ -98,20 +279,15 @@ routerboot_find_tag(u8 *buf, unsigned int buflen, u16 tag_id,
 	}
 
 	ret = -ENOENT;
-	while (buflen > 2) {
-		u16 id;
-		u16 len;
-
-		len = get_u16(buf);
-		buf += 2;
-		buflen -= 2;
-
-		if (buflen < 2)
-			break;
+	while (buflen > 4) {
+		u32 id_and_len = get_u32(buf);
+		buf += 4;
+		buflen -= 4;
+		id = id_and_len & 0xFFFF;
+		len = id_and_len >> 16;
 
-		id = get_u16(buf);
-		buf += 2;
-		buflen -= 2;
+		if (align)
```

Comment:
This is wrong. Moving this block here will break lzo decoding as the length passed to the subroutine will be incorrect.

Reply:
2 @f00b4r0 I am really bad in programming, I couldn't understand why placing here aligment condition could break lzo, if we are detecting RB_MAGIC_LZO lable in case condition align condition is ignored (align==false) so the lenght should be not changed?

Reply:
MAGIC_ERD uses lzo decompression. The comment below is in the `if (magic == RB_MAGIC_ERD) {` block.
Please note that robiemarko commented on the overall quality of the original code (from rbextract.c) here:
https://github.com/openwrt/openwrt/pull/2850#issuecomment-610277863